File: Task 1/Java/app/src/main/java/dev/ogabek/java/activity/MainActivity.java
package dev.ogabek.java.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.ProgressBar;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.gson.Gson;

import java.sql.Array;
import java.util.ArrayList;
import java.util.Arrays;

import dev.ogabek.java.R;
import dev.ogabek.java.adapter.Adapter;
import dev.ogabek.java.helper.Logger;
import dev.ogabek.java.model.Employee;
import dev.ogabek.java.network.VolleyHandler;
import dev.ogabek.java.network.VolleyHttp;

public class MainActivity extends AppCompatActivity {

    private RecyclerView rv_main;
    ProgressBar loading;
    FloatingActionButton btn_create;

    private final ArrayList<Employee> employees = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();
        get();

    }

    private void initViews() {
        rv_main = findViewById(R.id.rv_main);
        loading = findViewById(R.id.loading);
        btn_create = findViewById(R.id.btn_create);

        rv_main.setLayoutManager(new LinearLayoutManager((this)));

        btn_create.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                post();
            }
        });
    }

    public void dialogPoster(Employee employee) {
        new AlertDialog.Builder(MainActivity.this)
                .setTitle("Delete Poster")
                .setMessage("Are you sure you want to delete this poster?")
                .setPositiveButton("Yes", (dialogInterface, i) -> {
                    delete(employee.getId());
                })
                .setNegativeButton("No", null)
                .setIcon(R.drawable.ic_dialog_alert)
                .show();
    }

    private void get() {
        loading.setVisibility(View.VISIBLE);
        VolleyHttp.get(VolleyHttp.API_GET_ALL, VolleyHttp.paramsEmpty(), new VolleyHandler() {
            @Override
            public void onSuccess(String respond) {
                loading.setVisibility(View.GONE);
                Logger.d("VolleyHttp", respond);
                Employee[] employeesList = new Gson().fromJson(respond, Employee[].class);

                employees.clear();
                employees.addAll(Arrays.asList(employeesList));

                refreshAdapter(employees);
            }

            @Override
            public void onError(String error) {
                Logger.d("VolleyHttp", error);
                loading.setVisibility(View.VISIBLE);
            }
        });
    }

    private void post() {
        Employee employee = new Employee("OgabekDev", "18000", 18, "", 1);
        loading.setVisibility(View.VISIBLE);
        VolleyHttp.post(VolleyHttp.API_POST, VolleyHttp.paramsPost(employee), new VolleyHandler() {
            @Override
            public void onSuccess(String respond) {
                loading.setVisibility(View.GONE);
                Logger.d("VolleyHttp", respond);

                get();

                refreshAdapter(employees);
            }

            @Override
            public void onError(String error) {
                Logger.d("VolleyHttp", error);
                loading.setVisibility(View.VISIBLE);
            }
        });
    }

    private void delete(int id) {
        loading.setVisibility(View.VISIBLE);
        VolleyHttp.delete(VolleyHttp.API_DELETE + id, new VolleyHandler() {
            @Override
            public void onSuccess(String respond) {
                loading.setVisibility(View.GONE);
                Logger.d("VolleyHttp", respond);

                get();

                refreshAdapter(employees);
            }

            @Override
            public void onError(String error) {
                Logger.d("VolleyHttp", error);
                loading.setVisibility(View.VISIBLE);
            }
        });
    }

    private void refreshAdapter(ArrayList<Employee> employees) {
        Adapter adapter = new Adapter(this, employees);
        rv_main.setAdapter(adapter);
    }

}

File: Task 1/Kotlin/app/src/main/java/dev/ogabek/kotlin/activity/MainActivity.kt
package dev.ogabek.kotlin.activity

import android.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ProgressBar
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.gson.Gson
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.adapter.Adapter
import dev.ogabek.kotlin.helper.Logger
import dev.ogabek.kotlin.model.Employee
import dev.ogabek.kotlin.network.VolleyHandler
import dev.ogabek.kotlin.network.VolleyHttp
import java.util.*
import kotlin.collections.ArrayList

class MainActivity : AppCompatActivity() {

    private lateinit var rv_main: RecyclerView
    private lateinit var loading: ProgressBar
    private lateinit var btn_create: FloatingActionButton
    private val employees = ArrayList<Employee>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViews()
        get()
    }

    private fun initViews() {
        rv_main = findViewById(R.id.rv_main)
        loading = findViewById(R.id.loading)
        btn_create = findViewById(R.id.btn_create)
        rv_main.layoutManager = LinearLayoutManager(this)
        btn_create.setOnClickListener {
            post()
        }
    }

    fun dialogPoster(employee: Employee) {
        AlertDialog.Builder(this)
            .setTitle("Delete Poster")
            .setMessage("Are you sure you want to delete this poster?")
            .setPositiveButton(
                "Yes"
            ) { _, _ ->
                delete(employee.id)
            }
            .setNegativeButton("No", null)
            .setIcon(R.drawable.ic_dialog_alert)
            .show()
    }

    private fun get() {
        loading.visibility = View.VISIBLE
        VolleyHttp.get(VolleyHttp.API_GET_ALL, VolleyHttp.paramsEmpty(), object : VolleyHandler {
            override fun onSuccess(respond: String) {
                loading.visibility = View.GONE
                Logger.d("VolleyHttp", respond)
                val employeesList = Gson().fromJson(respond, Array<Employee>::class.java)
                employees.clear()
                employees.addAll(employeesList)
                refreshAdapter(employees)
            }

            override fun onError(error: String) {
                Logger.d("VolleyHttp", error)
                loading.visibility = View.VISIBLE
            }
        })
    }

    private fun post() {
        val employee = Employee("OgabekDev", "18000", 18, "", 1)
        loading.visibility = View.VISIBLE
        VolleyHttp.post(VolleyHttp.API_POST, VolleyHttp.paramsPost(employee), object : VolleyHandler {
            override fun onSuccess(respond: String) {
                    loading.visibility = View.GONE
                    Logger.d("VolleyHttp", respond)
                    get()
                    refreshAdapter(employees)
                }

            override fun onError(error: String) {
                    Logger.d("VolleyHttp", error)
                    loading.visibility = View.VISIBLE
                }
            })
    }

    private fun delete(id: Int) {
        loading.visibility = View.VISIBLE
        VolleyHttp.delete(VolleyHttp.API_DELETE + id, object : VolleyHandler {
            override fun onSuccess(respond: String) {
                loading.visibility = View.GONE
                Logger.d("VolleyHttp", respond)
                get()
                refreshAdapter(employees)
            }

            override fun onError(error: String) {
                Logger.d("VolleyHttp", error)
                loading.visibility = View.VISIBLE
            }
        })
    }

    private fun refreshAdapter(employees: ArrayList<Employee>) {
        val adapter = Adapter(this, employees)
        rv_main.adapter = adapter
    }
}